res/job_queue/job_queue_node.py

```diff
@@ -154,7 +154,10 @@ def _should_be_killed(self):
 
     def _job_monitor(self, driver, pool_sema, max_submit):
 
-        self.submit(driver)
+        submit_status = self.submit(driver)
+        if submit_status != 0:
+            self._set_status(JobStatusType.JOB_QUEUE_DONE)
```

Comment:
This makes a lot of sense given the transition schema for the job queue documented in the C code.